src/AutoMapper.Extensions.Microsoft.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -59,25 +58,25 @@ public static IServiceCollection AddAutoMapper(this IServiceCollection services,
         private static IServiceCollection AddAutoMapperClasses(IServiceCollection services, Action<IServiceProvider, IMapperConfigurationExpression> configAction, 
             IEnumerable<Assembly> assembliesToScan, ServiceLifetime serviceLifetime = ServiceLifetime.Transient)
         {
-            // Just return if we've already added AutoMapper to avoid double-registration
-            if (services.Any(sd => sd.ServiceType == typeof(IMapper)))
-                return services;
-
+            var sd = services.FirstOrDefault(sd => sd.ServiceType == typeof(IConfigurationProvider));

```

Comment:
This is what the Options stuff does, though. Why create our own factory, and leverage something out-of-the-box instead?

Reply:
It's not out of the box :) You have to add a package reference for how many lines of code, maybe five? I don't see the point.

Reply:
I don't mind a package reference in a library that is meant to support a specific package reference though, and in a way that is more or less the design of what this factory does.

Reply:
I think it just makes things more difficult to understand. Abstraction over clarity. My initial impression was that it might enable some client use cases, but I guess not.